### PR TITLE
Update specification with new models

### PR DESCRIPTION
**Pull request by adambedford**
This adds the following models:
- Tags
- Story tasks
- Project Templates
- Rate Cards
- Participations

And modifies the following:
- Workspace Group: Company fields
- Stories: `State` attribute for create/update
- Participations: Create and Update attributes

### CODE REVIEW

File: lib/mavenlink/story_task.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+module Mavenlink
+  class StoryTask < Model
+    # include Concerns::Indestructible
```
Comment (cantino) [inline lib/mavenlink/story_task.rb]: What is `Concerns::Indestructible`?

File: lib/mavenlink/tag.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+module Mavenlink
+  class Tag < Model
+    # include Concerns::Indestructible
```
Comment (cantino) [inline lib/mavenlink/tag.rb]: Here also.

File: lib/config/specification.yml [hunk 3]
```diff
@@ -318,6 +540,28 @@ story_allocation_days:
...
+    story:
+      foreign_key: story_id
+      collection: stories
+  attributes:
+    - id
+    - name
+    - completed
+    - completed_at
+    - position
+    - complted_by_id
```
Comment (cantino) [inline lib/config/specification.yml]: typo in `complted_by_id`

**Comment by cantino** [comment]
Thank you, this is super helpful!

**Comment by adambedford** [comment]
Fixed the typo and removed the unnecessary commented-out concern

**Comment by cantino** [comment]
Thanks!

**Pull request merged**